golem-worker-service-base/src/service/gateway/mod.rs

```diff
@@ -12,8 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use async_trait::async_trait;
+use golem_common::model::ComponentId;
+use golem_service_base::model::ComponentName;
 pub mod api_definition;
 pub mod api_definition_validator;
 pub mod api_deployment;
 pub mod http_api_definition_validator;
 pub mod security_scheme;
+
+#[async_trait]
+pub trait ConversionContext: Send + Sync {
```

Comment:
This is used to translate between our internal and public representation of components in the api definitions.